pkg/diagnose/diagnose.go

```diff
@@ -0,0 +1,48 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2017 Datadog, Inc.
+
+package diagnose
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/DataDog/datadog-agent/pkg/diagnose/diagnosis"
+	"github.com/cihub/seelog"
+	"github.com/fatih/color"
+)
+
+// Diagnose runs some connectivity checks, output it in writer
+// and returns the number of failed diagnosis
+func Diagnose(w io.Writer) (int, error) {
+	errorCounter := 0
+	if w != color.Output {
+		color.NoColor = true
+	}
+
+	// Use temporarily a custom logger to our Writer
+	oldLogger := seelog.Current
+	customLogger, err := seelog.LoggerFromWriterWithMinLevelAndFormat(w, seelog.DebugLvl, "[%LEVEL] %FuncShort: %Msg - %Ns%n")
+	if err != nil {
+		return -1, nil
+	}
+	seelog.UseLogger(customLogger)
+	defer seelog.ReplaceLogger(oldLogger)
```

Comment:
This looks a bit hacky but as all the other code is using the `Current` logger you have to swap it temporarily

Reply:
That's fine, let's expand that comment then, trying to foresee what implications might happen for other classes

Reply:
The major drawback that I can see is that in the flare you could potentially have other logs inserted in `diagnose.log` while the diagnose is running because the flare is run on the running agent through the API. It will not happen when you run the command on it's own because basically on the `diagnose` code is running so no chance to have non-related logs.
One way to avoid to avoid this if we want to keep leveraging logs would be to pass the custom logger to all the functions used in diagnostics, kind of a pain 😕